static/app/components/replays/breadcrumbs/replayTimelineSpans.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import type {RawSpanType} from 'sentry/components/events/interfaces/spans/types';
+import {divide, flattenSpans} from 'sentry/components/replays/utils';
+import Tooltip from 'sentry/components/tooltip';
+import {t, tct} from 'sentry/locale';
+import space from 'sentry/styles/space';
+
+type Props = {
+  /**
+   * Duration, in milliseconds, of the timeline
+   */
+  duration: number;
+
+  /**
+   * The spans to render into the timeline
+   */
+  spans: RawSpanType[];
+
+  /**
+   * Timestamp when the timeline begins, in milliseconds
+   */
+  startTimestamp: number;
+
+  /**
+   * Extra classNames
+   */
+  className?: string;
+};
+
+function ReplayTimelineEvents({className, duration, spans, startTimestamp}: Props) {
+  const flattenedSpans = flattenSpans(spans);
+
+  const startMs = startTimestamp * 1000;
+  return (
+    <Spans className={className}>
+      {flattenedSpans.map(span => {
+        const sinceStart = span.startTimestamp - startMs;
+        const startPct = divide(sinceStart, duration);
+        const widthPct = divide(span.duration, duration);
+
+        const requestsCount =
+          span.span_count === 1
+            ? t('1 network request')
+            : tct('[count] network requests', {count: span.span_count});
+        return (
+          <Tooltip
+            key={span.span_id}
+            title={
+              <React.Fragment>
+                {requestsCount}
+                <br />
+                {span.duration.toFixed(2)}ms
+              </React.Fragment>
+            }
+            skipWrapper
+            disableForVisualTest
+          >
+            <Span startPct={startPct} widthPct={widthPct} />
+          </Tooltip>
+        );
+      })}
+    </Spans>
+  );
+}
+
+const Spans = styled('ul')`
+  /* Reset defaults for <ul> */
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  height: ${space(3)};
+  position: relative;
+`;
+const Span = styled('li')<{startPct: number; widthPct: number}>`
+  display: block;
+  position: absolute;
+  top: 0;
+  left: ${p => p.startPct * 100}%;
+  min-width: 1px;
+  width: ${p => p.widthPct * 100}%;
+  height: 100%;
+  background: #865189; /* plucked from static/app/constants/chartPalette.tsx */
```

Comment:
Should we import this value so that it remains synced with our chart palette?

Reply:
idk how, the pallet is a giant array and I havn't looked into how colors are picked in the network spans component.

Our latest designs show different colors used for different types of spans in the timeline, so that'll be something to follow up on